<div id="navegacion" prev="../Tema_4/t4_funciones_02_pdi.html" 
                     next="../Ejercicios_Tema_3/ej2.html"></div>

<div class="section-title">Section 2.8</div>

# Exercises

## Exercise 1

Write the Python code for the following expression:

$$\frac{3}{4} + 2 * ( 5^4 - \frac{1}{2} )$$



In [2]:
# Sol:
(3/4)+2*(5**4-1/2)


1249.75

Write a Python function named __expression_value__ that will return the value of following expression:

$$\frac{3}{4} + 2 * ( x^4 - \frac{1}{2} )$$

for a given value of $x$. The function has a single input argument of numerical type and returns the result of applying the previous expression.

In [4]:
# Sol

def expresion_value(x):
    return (3/4)+2*(x**4-1/2)


Run the function __expression_value__ for the values 5 and 3. 

__Remark:__ 

* `expression_value(5)` returns `1249.75`
* `expression_value(3)` returns `161.75`

In [6]:
# Sol
expresion_value(5)
expresion_value(3)

161.75

## Exercise 2

Define a function named __cubic_square__ that will return the value of $\sqrt[3]{x}$. Remember that $\sqrt[3]{x}$ = ${x}^{\frac{1}{3}}$


__Remark:__ 

* `cubic_square(5)` returns `1.7099759466766968`

In [8]:
# Sol:
def cubic_square(x):
    return x**(1/3)

cubic_square(5)

1.7099759466766968

## Exercise 3

Write a function named __inside__ that receives two input parameters. The first parameter is a list of elements and the second one is an element. The function returns `True` if the element is in the list. Otherwise returns `False`. 

* You can use the `if-else` statement
* You can use the function `in` to know if an element is inside the sequence

__Remark:__

* `inside([1, 3, 6], 1)` returns `True`
* `inside([1, 3, 6], 5)` returns `False`
* `inside([1,'Hi', 6, 'Hola'] , 'Hola')` returns `True`

In [3]:
# Sol:
def inside(list, element):
    for i in list:
        if i == element:
            return True
    return False

    
print(inside([1,3,6],1))
print(inside([1,3,6],5))
print(inside([1,'Hi',6,'Hola'],'Hola'))



True
False
True


## Exercise 4

The `upper` function for strings (`str`) creates a new string, where all the characters are capital letters.
Run the next cell to see the effect of the `upper` function when applying it to the string `s`:

In [4]:
s = 'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'
s.upper()

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

Look that `upper` does not modify the original string (strings are immutable), but returns a new one.

In [5]:
s    # string s is unchanged

'Hola, me llamo Iñigo Montoya, tú mataste a mi padre. ... Prepárate a morir'

In order to save the `upper` result, we must use an assignment:

In [6]:
t = s.upper()

We can ask for the content in `t`:

In [7]:
t

'HOLA, ME LLAMO IÑIGO MONTOYA, TÚ MATASTE A MI PADRE. ... PREPÁRATE A MORIR'

* Create a string `t`  that is equal to `s`  but in lower case (`lower` function). Compute the length of `t`.
* Run the `split(',')` function. Take a look at what the function returns. 
* Run the `title()` function.
* Run the `replace(' ', '_')` function. Take a look at what the function returns
* Run `dir(str)` in order to list all the available functions for strings.

In [19]:
# Sol: 
t = s.lower()
print(t)
print(len(t))
t.split(',')
t.title()
t.replace(' ','_')

#resturn list all the avaiable funtions for strings
dir(t)




hola, me llamo iñigo montoya, tú mataste a mi padre. ... prepárate a morir
74


['__add__',
 '__class__',
 '__contains__',
 '__delattr__',
 '__dir__',
 '__doc__',
 '__eq__',
 '__format__',
 '__ge__',
 '__getattribute__',
 '__getitem__',
 '__getnewargs__',
 '__gt__',
 '__hash__',
 '__init__',
 '__init_subclass__',
 '__iter__',
 '__le__',
 '__len__',
 '__lt__',
 '__mod__',
 '__mul__',
 '__ne__',
 '__new__',
 '__reduce__',
 '__reduce_ex__',
 '__repr__',
 '__rmod__',
 '__rmul__',
 '__setattr__',
 '__sizeof__',
 '__str__',
 '__subclasshook__',
 'capitalize',
 'casefold',
 'center',
 'count',
 'encode',
 'endswith',
 'expandtabs',
 'find',
 'format',
 'format_map',
 'index',
 'isalnum',
 'isalpha',
 'isascii',
 'isdecimal',
 'isdigit',
 'isidentifier',
 'islower',
 'isnumeric',
 'isprintable',
 'isspace',
 'istitle',
 'isupper',
 'join',
 'ljust',
 'lower',
 'lstrip',
 'maketrans',
 'partition',
 'removeprefix',
 'removesuffix',
 'replace',
 'rfind',
 'rindex',
 'rjust',
 'rpartition',
 'rsplit',
 'rstrip',
 'split',
 'splitlines',
 'startswith',
 'strip',
 'swapcase',


--------------------